pkg/controller/core/clusterqueue_controller.go

```diff
@@ -498,3 +499,17 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus() *kueue.ClusterQueuePendingWorkloadsStatus {
+	pendingWorkloads := make([]kueue.ClusterQueuePendingWorkload, 0)
+	for _, workloads := range r.qManager.GetSnapshots() {
+		pendingWorkloads = append(pendingWorkloads, workloads...)
```

Comment:
This looks suspicious. IIUC the qManager holds the workloads keyed by CQ name. Here we want to return pendingWorkloads corresponding to only the given cluster queue. I think instead of calling `GetSnapshots` we should have `GetSnapshot(cq)`

Reply:
https://github.com/kubernetes-sigs/kueue/pull/1069/files#diff-54fb2e1e38c1d823a077207b574c6dc9f21d0b354584becd6940b26282aec41fR550